### PR TITLE
split the web api/web app service collection extention methods and cr…

### PR DESCRIPTION
**Pull request by jennyf19**
…eated ones for the builder in a separate class to simplify the api

update unit tests

fix for #78

### CODE REVIEW

**Comment by jennyf19** [comment]
New Api diagrams:
image

image

image

image

File: src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -36,111 +32,21 @@ public static class WebApiServiceCollectionExtensions
...
         {
+            AuthenticationBuilder builder = services.AddAuthentication(jwtBearerScheme);
             return builder.AddProtectedWebApi(options => configuration.Bind(configSectionName, options),
                 options => configuration.Bind(configSectionName, options),
                 tokenDecryptionCertificate,
                 jwtBearerScheme,
                 subscribeToJwtBearerMiddlewareDiagnosticsEvents);
         }
 
-        /// <summary>
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs:53]: (Moved to WebApiAuthenticationBuidlerExtensions.cs)

**Comment by jmprieur** [comment]
@pmaytak : actually, Jenny asked me and I though we'd only add them if customers ask for them

**Comment by jennyf19** [comment]
Thanks for the feedback...I'll merge this and we can discuss the above. :)

**Comment by pmaytak** [comment]
My reasoning:
- Overloads with action parameters give more flexibility and power to the end user. (That's why we added them, as I understand.)
- Methods extending `ServiceCollection` simplify and allow the user to not call `AddAuthentication` methods every time.
- Other extension methods already have the two types of overloads. Adding these two will contribute to the consistency of the API.
- We can simplify _Startup_ classes in our samples to use the `ServiceCollection` extensions, like so:
```CSharp
...
// Adds Microsoft Identity platform (AAD v2.0) support to protect this Api
services.AddProtectedWebApi(Configuration, "AzureAd");
...
```
- Lastly, I think we don't even have to call `AddAuthentication(JwtBearerDefaults.AuthenticationScheme)` and `AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)` methods at all because in the configuration methods we eventually call `builder.AddJwtBearer(jwtBearerScheme, options => ...);` and `builder.AddOpenIdConnect(openIdConnectScheme, options => ... );`, which set the same authentication scheme anyway. I am not a 100% sure on this; it's just a hunch based on the Core source code I saw. So we would need to create a separate issue for this.

**Comment by jmprieur** [comment]
Thanks @pmaytak 
- For the last bullet point it's worth asking George. Do you want to do it?
- For the bullet point before last, this is already the case?  they are simple already for the simple scenario.

I prefer that we work for now on making sure the public API has the shape we want. We can add the other methods later.

**Comment by pmaytak** [comment]
- I know where the code for that is, but I'll verify with George. I created #83 so I won't lose track of this.
- Below is how it is right now. I agree that it's simple already, but there's always room for improvement.
```CSharp
// Adds Microsoft Identity platform (AAD v2.0) support to protect this Api
services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
       .AddProtectedWebApi(Configuration, "AzureAd");
```
- That's fine, as long as we had a discussion on this.

**Pull request merged**